internal/topology/names/names.go

```diff
@@ -106,6 +106,16 @@ func MachineSetMachineNameGenerator(templateString, clusterName, machineSetName
 		})
 }
 
+// InfraClusterNameGenerator returns a generator for creating a infrastructure cluster name.
+func InfraClusterNameGenerator(templateString, clusterName, infraClusterName string) NameGenerator {
+	return newTemplateGenerator(templateString, clusterName,
+		map[string]interface{}{
+			"infraCluster": map[string]interface{}{
+				"name": infraClusterName,
+			},
+		})
+}
```

Comment:
```suggestion
// InfraClusterNameGenerator returns a generator for creating a infrastructure cluster name.
func InfraClusterNameGenerator(templateString, clusterName string) NameGenerator {
	return newTemplateGenerator(templateString, clusterName,
		map[string]interface{}{})
}
```
This should look like this.

In my opintion we should not pass in the name of the InfraClusterTemplate and then provide it here as `infraCluster.name` to templating